python/lsst/pipe/base/pipeline.py

```diff
@@ -263,16 +299,22 @@ def subsetFromLabels(self, labelSpecifier: LabelSpecifier) -> Pipeline:
         return Pipeline.fromIR(self._pipelineIR.subset_from_labels(labelSet))
 
     @staticmethod
-    def _parseFileSpecifier(fileSpecifer):
+    def _parseFileSpecifier(fileSpecifer) -> Tuple[str, Optional[LabelSpecifier]]:
         """Split appart a filename path from label subsets
         """
-        split = fileSpecifer.split(':')
+        # This is to support legacy pipelines during transition
+        split = re.split("[:](?!\\/\\/)", fileSpecifer)
+        if len(split) > 1:
+            warnings.warn("The fileSpecifier seems to use the legacy : to separate labels, this is "
+                          "deprecated and will be removed in June 2021, please use # instead.")
```

Comment:
A user that sees this warning is unlikely to know what "fileSpecifier" means, because that's sort of specific to this code context.   Maybe "The pipeline file {fileSpecifier!r} seems to use ..." instead?

You should probably also make this a `DeprecationWarning` or `FutureWarning`, rather than the default `UserWarning` (I don't recall the exact syntax, or which of those the dev guide recommends).